packages/migrate/src/__tests__/DbPull.test.ts

```diff
@@ -481,6 +481,84 @@ describe('postgresql - missing database', () => {
   })
 })
 
+describe('postgresql views fs I/O', () => {
+  const connectionString = process.env.TEST_POSTGRES_URI_MIGRATE!.replace('tests-migrate', 'tests-migrate-db-pull')
+
+  const fixturePath = path.join('introspection', 'postgresql', 'views-fs')
+
+  const setupParams: SetupParams = {
+    connectionString,
+    // Note: dirname points to a location with a setup.sql file
+    // which will be executed to prepare the database with the correct tables, views etc.
+    dirname: path.join(__dirname, '..', '__tests__', 'fixtures', fixturePath),
+  }
+
+  beforeEach(async () => {
+    await setupPostgres(setupParams)
+
+    // Back to original env vars
+    process.env = { ...originalEnv }
+    // Update env var because it's the one that is used in the schemas tested
+    process.env.TEST_POSTGRES_URI_MIGRATE = connectionString
+  })
+
+  afterEach(async () => {
+    // Back to original env vars
+    process.env = { ...originalEnv }
+    await tearDownPostgres(setupParams).catch((e) => {
+      console.error(e)
+    })
+  })
+
+  test('introspection creates view definition files', async () => {
+    ctx.fixture(path.join(fixturePath))
+
+    const introspect = new DbPull()
+    const result = introspect.parse([])
```

Comment:
Other things to check in this test:
- log/warn/error snapshots

Other scenarios to check, using this test as a template:
- additional files in the `views` folder that are not relevant for the introspected view definitions should be deleted after `prisma db pull`
- same tests with non-standard prisma paths (the `views` folder is defined exactly one level prior to the `prisma.schema` file in the directory tree)
- any other useful expectation that comes to your mind

Reply:
> same tests with non-standard prisma paths (the views folder is defined exactly one level prior to the prisma.schema file in the directory tree)

I'm honestly not sure whether this is trying to say that the views folder _is_ defined as such, or that that would be a non-standard path and an example to test 🤔

Reply:
this is the intended standard, no? so then testing when `/views/` exists on the same dir level as `/prisma/`
<img width="175" alt="image" src="https://user-images.githubusercontent.com/29753584/226995127-f6a90076-d40f-4438-9c73-0a63b18fe1d5.png">

Reply:
The `views` folder should be next to the schema file (which by default is `prisma/schema.prisma` - but can also be `schema.prisma` or any other filename on any path if customized via https://www.prisma.io/docs/concepts/components/prisma-schema#prisma-schema-file-location

Reply:
Some ideas for better and more extensive tests:
- Test some/all variants for schema file location from https://www.prisma.io/docs/concepts/components/prisma-schema#prisma-schema-file-location
- Snapshot the full Introspection result like in the basic introspection test so we have a full text representation of the returned values in the test file
- Test without preview feature to confirm nothing changes in the files
- also test other databases (all SQL should be supported) with one trivial case at least that should create a file